### PR TITLE
Added eager loading for `changedinfo` and removed encrypted values from Asset history

### PR DESCRIPTION
**Pull request by Godmartinz**
# Description

Checks if a field is encrypted and returns `encrypted` instead of the hashed string. asset_eol_date now reads `EOL date` in the Asset History Log

image

Also eager loads the queries with soft deletes which was needed for the `changedInfo` method

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* PHP version:
* MySQL version
* Webserver version
* OS version

# Checklist:

- [ ] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [ ] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Godmartinz** [comment]
added soft deletes to all the queries:
` $location = Location::withTrashed()->get();
        $company = Company::withTrashed()->get();
        $supplier = Supplier::withTrashed()->get();
        $model = AssetModel::withTrashed()->get();`

File: app/Http/Transformers/ActionlogsTransformer.php [hunk 1]
```diff
@@ -143,39 +156,49 @@ public function transformCheckedoutActionlog (Collection $accessories_users, $to
      * @param  array $clean_meta
      * @return array
      */
-    public function changedInfo(array $clean_meta)
-    {
+    public function changedInfo(array $clean_meta) {
+
+        $location = Location::withTrashed()->get();
+        $company = Company::withTrashed()->get();
```
Comment (marcusmoore) [inline app/Http/Transformers/ActionlogsTransformer.php]: This fails for me since `Company` doesn't have the `SoftDeletes` trait on it.

**Review by marcusmoore — CHANGES_REQUESTED** [review]

File: app/Http/Transformers/ActionlogsTransformer.php (hunk 1 above)
Comment (snipe) [reply app/Http/Transformers/ActionlogsTransformer.php]: Same (and good catch)

Comment (marcusmoore) [reply app/Http/Transformers/ActionlogsTransformer.php]: @Godmartinz I think the lines starting at 168 in [this branch](https://github.com/snipe/snipe-it/compare/master...marcusmoore:snipe-it:bug/sc-23635#diff-7fe056d76c09808dac923c4639161d587c3fff281a01122f3e10c4a781674a65R168-R175) should handle this for companies.

Comment (Godmartinz) [reply app/Http/Transformers/ActionlogsTransformer.php]: added the fix

**Pull request merged**